pkg/odo/cli/logs/logs.go

```diff
@@ -173,14 +200,17 @@ func getUniqueContainerName(name string, uniqueNames map[string]struct{}) string
 }
 
 // printLogs prints the logs of the containers with container name prefixed to the log message
-func printLogs(containerName string, rd io.ReadCloser, out io.Writer) error {
-	color.Set(log.ColorPicker())
-	defer color.Unset()
+func printLogs(containerName string, rd io.ReadCloser, out io.Writer, colour color.Attribute, mu *sync.Mutex) error {
 	scanner := bufio.NewScanner(rd)
 	scanner.Split(bufio.ScanLines)
 
 	for scanner.Scan() {
 		line := scanner.Text()
+		mu.Lock()
+		defer mu.Unlock()
```

Comment:
But with this, you are locking for each iteration of the `for` loop, but unlocking when the function returns. Is a `sync.Mutex` reentrant? If not, I guess this call would deadlock or error.

Reply:
Maybe this could be done as you did, but in a separate function handling each line..

Reply:
or using a lambda:

```
line := scanner.Text()
func() {
  mu.Lock()
  defer mu.Unlock()
  ...
/* exactly what you want to lock here */
}()

Reply:
> But with this, you are locking for each iteration of the `for` loop, but unlocking when the function returns. Is a `sync.Mutex` reentrant? If not, I guess the call to `mu.Lock()` at the next iteration would deadlock or error.

Totally missed that I'm doing this inside a `for` loop. Friday effect, maybe. 😉 
@feloy I like the anonymous function idea. PTAL 084d0f4.